### PR TITLE
Added NPM Workspaces

### PR DESCRIPTION
**Pull request by mqnguyen5**
## Issue This PR Addresses

Introduced [npm workspaces](https://docs.npmjs.com/cli/v8/using-npm/workspaces) to help manages dependencies in `/api` and `/web` and also allow for single `npm i` through the root directory

## Type of Change

- [ ] **Bugfix**: Change which fixes a bug
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

- Renamed root `/src` directory into `/workspaces` for better differentiation
- Added basic configuration for workspaces in root `package.json`

## Steps to test the PR

1. Navigate to SplitWise root directory
2. Run `npm i`
3. Change into `/api` and run `npm run dev`
4. Open the browser and navigate to `localhost:6060/api/utils/healthcheck`
5. Inspect the response JSON

## Checklist

- [ ] **Test**: make sure this PR passes GitHub action checks
- [ ] **Screenshots**: pictures/vidoes/GIF to demo this PR
- [ ] **Documentation**: this PR includes documentation of new changes, could be comments in code or documentation files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
   "name": "splitwise3",
+  "workspaces": [
+    "./workspaces/api",
+    "./workspaces/web"
```
Comment (TueeNguyen) [inline package.json:5]: Let's not change `src` to `workspaces`
```suggestion
"./src/api",
"./src/web"
```

**Review by TueeNguyen — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mqnguyen5) [reply package.json:5]: alright, let me revert it back

File: src/api/src/firebase/configs.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ require('dotenv').config();
 let adminCredentials = {}; // firebase-admin service account credentials
 let firebaseCredentials = {}; // firebase SDK configuration
 
-process.env.NODE_ENV === process.env.NODE_ENV || 'development';
+process.env.NODE_ENV = process.env.NODE_ENV || 'development';
```
Comment (TueeNguyen) [inline src/api/src/firebase/configs.ts:6]: nice

Comment (mqnguyen5) [reply src/api/src/firebase/configs.ts:6]: yeah, I also fixed that as well since we discovered it in the call

**Review by TueeNguyen — APPROVED** [review]

**Pull request merged**